### PR TITLE
Fixes for the ESLint vscode intergration

### PR DESCRIPTION
**Pull request by jcompagner**
set the right nodePath (should be a (parent) dir that has the
node_modules that has again eslint)
set the workingFolder (for now the highest folder with a package.json)
and set the workingDirectory property in auto mode so eslint server will
try to find the best location of the eslint config file for the given
file.

### CODE REVIEW

**Comment by mickaelistria** [comment]
That looks quite good, and I'm eager to merge this!
See some comments inline about using more "size-aware" and more immutable constructors; that will save a few bytes on every call, ultimately improving performance.
Also, please `git commit --amend --signoff` and then `git push --force`your commit to add the necessary Eclipse ECA line.

**Comment by jcompagner** [comment]
which comments inline are you talking about?
maybe i look in the wrong place but i don't see any comments added to the commit above as far as i can see..

i guess the size-aware things are the 2 or 3 hashmaps constructors (we know we only add certain amount of keys)
but not sure what part of code is about " more immutable constructors;"

**Comment by mickaelistria** [comment]
Sorry, I forgot to push the review... Too many buttons!

**Review by mickaelistria — CHANGES_REQUESTED** [review]

File: org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/eslint/ESLintClientImpl.java [hunk 1]
```diff
@@ -28,10 +31,51 @@
...
+
+		// this should not point to a nodejs executable but nodePath is the path to the "node_modules" 
+		// (or a parent having node modules, we just push in the highest dir that has the package.json)
+		config.put("nodePath", highestPackageJsonDir.getAbsolutePath());
+		
 		config.put("validate", "on");
 		config.put("run", "onType");
-		config.put("nodePath", NodeJSManager.getNodeJsLocation().getAbsolutePath());
+
+		Map<String, Object> codeAction = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (jcompagner) [reply org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/eslint/ESLintClientImpl.java:72]: thx for teaching me that api ;)
i really need to get working on code higher then 8 i see (our current product is still fully on 8, try to move to java 11 but that is not so easy with a lot of dependencies..)

**Comment by jcompagner** [comment]
i see you also have the same problem i have the last few days

jarsigner: unable to sign jar: java.io.IOException: Server returned HTTP response code: 400 for URL: http://sha256timestamp.ws.symantec.com/sha256/timestamp

that service is very flaky, our builds are also now and then failing with that...

**Comment by mickaelistria** [comment]
Thanks a lot @jcompagner !

**Pull request merged**